### DISCUSSION TITLE
Built-in zoom controls?

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#912] [Q&A] [answered]
I want to support dedicated zoom buttons on my map. (I know that "zoom by pinch" is supported, but sometimes "a real" button is easier to handle for the end-user.)

Mapsforge lib supports built-in zoom controls with some configuration options, e. g. position, drawables etc.

Couldn't find any built-in zoom controls for VTM, though, not in the examples and not in the source. Have I missed something, or do zoom controls have to be built manually (adding a layer with two drawables, attaching tap controllers to it, linked to zooming in/out)?

**Comment by devemux86** [comment, marked as answer]
Mapsforge had the now deprecated Android [ZoomButton](https://developer.android.com/reference/android/widget/ZoomButton).

---

They are libraries (not apps), so no one should have and maintain buttons.

Each app usually has different needs regarding appearance, size, position, etc.
VTM map on Android is a View, you can add on top anything, like zoom buttons.

**Reply by moving-bits** [reply]
Ok, thanks. We've been using Mapsforge's `MapZoomControls` in the past, have switched to adding zoom buttons on top of the map view for the new implementation, works fine.
